### PR TITLE
Set KOLIBRI_LISTEN_PORT default to 8080 and use it in cli.start()

### LINKED ISSUE #1724: KOLIBRI_LISTEN_PORT doesn't seem to get used
We currently have alternate methods for setting the port, and this envvar doesn't seem to get used at all, so should we remove it? Or use it?

```
Searching 14964 files for "KOLIBRI_LISTEN_PORT" (case sensitive)

/home/jamalex/fle/kolibri/kolibri/utils/cli.py:
   26      "KOLIBRI_HOME", os.path.join(os.path.expanduser("~"), ".kolibri")
   27  )
   28: os.environ.setdefault("KOLIBRI_LISTEN_PORT", "8008")
   29  
   30  import django  # noqa

1 match in 1 file
```

### PR DESCRIPTION
**Pull request by benjaoming**
## Summary

FIxes #1724

## TODO

- [x] Have tests been written for the new code?
- [x] Has documentation been written/updated?
- [x] New dependencies (if any) added to requirements file
- [x] Add an entry to CHANGELOG.rst
- [x] Add yourself it AUTHORS.rst if you don't appear there

## Issues addressed

#1724

## Documentation

Looking through docs to see if there's any changes to be made...

### CODE REVIEW

**Comment by benjaoming** [comment]
Looking into a few things to understand the loss of test coverage, added a more deliberate test of `update_channel_metadata_cache` although it doesn't actually check that data is correctly annotated :/ I'm not sure if that sort of coverage is super nice, but alternatively, I'd suggest to add an issue about it.

**Comment by rtibbles** [comment]
And another merge conflict resolved!

**Comment by rtibbles** [comment]
Except I broke everything. Sadness :(

**Comment by benjaoming** [comment]
Working on it now!

**Comment by benjaoming** [comment]
Some strange stuff going on with the postgres tests:

```
kolibri/utils/tests/test_cli.py:73: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
kolibri/utils/cli.py:237: in start
    update()
kolibri/utils/cli.py:211: in update
    call_command("migrate", interactive=False, database="ormq")
.tox/postgres/lib/python2.7/site-packages/django/core/management/__init__.py:119: in call_command
    return command.execute(*args, **defaults)
.tox/postgres/lib/python2.7/site-packages/django/core/management/base.py:399: in execute
    output = self.handle(*args, **options)
.tox/postgres/lib/python2.7/site-packages/django/core/management/commands/migrate.py:66: in handle
    connection = connections[db]
.tox/postgres/lib/python2.7/site-packages/django/db/utils.py:209: in __getitem__
    self.ensure_defaults(alias)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <django.db.utils.ConnectionHandler object at 0x7fa9760f3050>
alias = 'ormq'
    def ensure_defaults(self, alias):
        """
            Puts the defaults into the settings dictionary for a given connection
            where no settings is provided.
            """
        try:
            conn = self.databases[alias]
        except KeyError:
>           raise ConnectionDoesNotExist("The connection %s doesn't exist" % alias)
E           ConnectionDoesNotExist: The connection ormq doesn't exist

```

**Comment by benjaoming** [comment]
Rebased and force-pushed. Please have patience.. will figure out what's up with the postgres test.

**Comment by benjaoming** [comment]
Rebased to upstream to resolve conflicts, seems ready! @rtibbles

**Comment by benjaoming** [comment]
Seemed to still fail in the postgres tests (`kolibri/core/webpack/test/test_webpack_tags.py::TestHook: cannot collect test class 'TestHook' because it has a __init__ constructor`), so I changed some things

File: kolibri/utils/server.py [hunk 1]
```diff
@@ -61,6 +61,12 @@ def start(port=8080):
     with open(PID_FILE, 'w') as f:
         f.write("%d\n%d" % (os.getpid(), port))
 
+    # TODO: not sure where it should ideally be located. Question is if it
+    # should be run every time the server is started or if it depends on some
+    # kind of state change.
```
Comment (jamalex) [inline kolibri/utils/server.py]: It gets triggered when state changes via the app itself, but state could also be changed by copying a `content` folder into `.kolibri`, or deleting a channel DB on disk, so it's best to have an extra call on start. It should be a pretty fast thing.

Comment (benjaoming) [reply kolibri/utils/server.py]: Thanks, I've updated the comment!

**Comment by benjaoming** [comment]
Notice that I actually _improved_ test coverage even though in an area where it was already very low :)

**Comment by benjaoming** [comment]
@jamalex @rtibbles please merge at will, I have nothing more to add here, will pick up on test coverage as #1595 is implemented.

**Comment by benjaoming** [comment]
The PR also removes creation of a legacy `ormq` database which belonged to django-q.

**Pull request merged**